### ISSUE TITLE
appstream-util: validate-strict doesn't detect critical typos

### ISSUE THREAD

**Issue by BenWiederhake** [hughsie/appstream-glib#77]
Example file:

	<?xml version="1.0" encoding="utf-8"?>
	
	<component type="addon">
		<id>telegram-purple</id>
		<extends>pidgin.desktop</extends>
		<name>Telegram</name>
		<name xml:lang="de-DE">Telegram</name>
		<summary>A plugin that adds support for Telegram</summary>
		<summary xml:lang="de-DE">Plugin zur Unterstützung von Telegram</summary>
		<url type="homepage">https://github.com/majn/telegram-purple</url>
		<url type="bugtracker">https://github.com/majn/telegram-purple/issues/</url>
		<metadata_license>GFDL-1.3</metadata_license>
		<project_license>GPL-2.0+</project_license>
		<updatecontact>eischmann_at_redhat.com</updatecontact>
	</component>

Expected output (taken from [`appstreamcli validate`](https://packages.debian.org/stretch/appstream))

	W - telegram-purple.metainfo.xml:telegram-purple
		Found invalid tag: 'updatecontact'. Non-standard tags have to be prefixed with "x-".

	Validation failed.

Actual output:

	AppStream/telegram-purple.metainfo.xml: OK

The underlying error in the file is that I misspelled [`update_contact`](http://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-update_contact) (with underscore) as `updatecontact` (without underscore).

I believe `appstream-util validate-strict` should at least warn me about typos that break the AppStream datafile. Please implement this.

**Comment by hughsie** [comment]
This isn't really a bug; let me explain: When we came up with AppStream specification we implemented two libraries, libappstream and libappstream-glib -- with slightly different focuses. I wanted to add `updatecontact` into the spec and so added it to the appdata page as an unofficial tag when the project started which thousands of people looked at. During this time over 500 projects added AppData files, with the crucial updatecontact tag. Mattias agreed to add updatecontact to the specification, but did so (probably correctly) as `update_contact` -- but we had a few hundred existing files in the wild with the old tag name.

libappstream-glib accepts either form, both old and new. Quite a few people use "appstream-util validate" in the build and CI, and so making a file suddenly start failing is somewhat antisocial. I think the pragmatic thing to do is still accept either form, although I also accept that `appstreamcli validate` is sticking to the specification and that it's strictly doing the right thing.

**Comment by BenWiederhake** [comment]
This **is** a bug, because a conforming program that wants to display AppStream data now fails to do so, and has to support undocumented, broken, historic behavior. Down that path lies RSS.

**Comment by BenWiederhake** [comment]
Sorry, this wasn't a constructive comment. Let me try again.

Yeah, it shouldn't start failing suddenly. However, it should start converging towards the *single* specification, which could be done like this:

- now: `validate-strict` starts outputting a warning, but returns "all good"
- three months (or whatever time interval seems appropriate): `validate-strict` treats it as an error, and returns "failure". Meanwhile, `validate` starts outputting a warning, but still returns "all good".
- in a year: `validate` considers it an error, too, and `validate-relax` says it's a warning.

How about this? This would indeed start "breaking" everywhere, but since it's an actual problem, I think this is a good thing. Also, it can be fixed temporarily by dialling down the strictness. All are happy, and the world converges towards a *single* specification of AppStream.
